__tests__/__utils__/handlers.ts

```diff
@@ -90,41 +91,55 @@ export function createUuidHandler(uuid: Model<'AbstractUuid'>, once?: boolean) {
 
 export function createMessageHandler(
   args: {
-    message: {
-      type: string
-      payload?: Record<string, unknown>
-    }
+    message: MessagePayload
     body?: unknown
     statusCode?: number
   },
   once = false
 ) {
-  const { message, body } = args
+  const { message, body, statusCode } = args
 
-  const handler = rest.post(
-    getDatabaseLayerUrl({ path: '/' }),
-    (_req, res, ctx) => {
+  return createDatabaseLayerHandler({
+    matchMessage: message,
+    resolver: (_req, res, ctx) => {
       return (once ? res.once : res)(
-        ctx.status(args.statusCode ?? 200),
+        ctx.status(statusCode ?? 200),
         ...(body === undefined
           ? []
           : [ctx.json(body as Record<string, unknown>)])
       )
-    }
-  )
+    },
+  })
+}
+
+export function createDatabaseLayerHandler<Payload = DefaultPayloadType>(args: {
+  matchMessage: MessagePayload<Payload>
+  resolver: RestResolver<Required<MessagePayload<Payload>>>
+}) {
+  const { matchMessage, resolver } = args
+
+  const handler = rest.post(getDatabaseLayerUrl({ path: '/' }), resolver)
 
   // Only use this handler if message matches
-  handler.predicate = (req) => {
-    return R.equals(req.body, message)
-  }
+  handler.predicate = (req) =>
+    req.body.type === matchMessage.type &&
+    (matchMessage.payload === undefined ||
+      R.equals(req.body.payload, matchMessage.payload))
 
   return handler
 }
 
-export function getDatabaseLayerUrl({ path }: { path: string }) {
+function getDatabaseLayerUrl({ path }: { path: string }) {
   return `http://${process.env.SERLO_ORG_DATABASE_LAYER_HOST}${path}`
 }
 
+interface MessagePayload<Payload = DefaultPayloadType> {
+  type: string
+  payload?: Payload
+}
```

Comment:
I don't see the type generic used anywhere (and it probably won't even work correctly since `createMessageHandler` always uses the fallback type generic `DefaultPayloadType`. So I think this could be

```suggestion
interface MessagePayload {
  type: string
  payload?: Record<string, unknown>
}
```
instead and simplify some of the other function signatures, right?

Reply:
The type generics is used for example at https://github.com/serlo/api.serlo.org/blob/f18167d05e85faff1abc53390ed8120db8573548/__tests__/schema/thread/thread.ts#L440-L448 Here we have `Payload = { id: number }` so that `req.body.payload.id` can be requested without any complains of typescript.